ReplCommand.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (C) 2020 Grakn Labs
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package grakn.console;
+
+import grakn.client.Grakn;
+import org.jline.reader.EndOfFileException;
+import org.jline.reader.LineReader;
+import org.jline.reader.UserInterruptException;
+
+public abstract class ReplCommand {
+    static class Exit extends ReplCommand {}
+    static class Help extends ReplCommand {}
+    static class Clear extends ReplCommand {}
+    static class DatabaseList extends ReplCommand {}
+    static class DatabaseCreate extends ReplCommand {
+        private final String database;
+        public DatabaseCreate(String database) {
+            this.database = database;
+        }
+        public String database() { return database; }
+    }
+    static class DatabaseDelete extends ReplCommand {
+        private final String database;
+        public DatabaseDelete(String database) {
+            this.database = database;
+        }
+        public String database() { return database; }
+    }
+    static class Transaction extends ReplCommand {
+        private final String database;
+        private final Grakn.Session.Type sessionType;
+        private final Grakn.Transaction.Type transactionType;
+        public Transaction(String database, Grakn.Session.Type sessionType, Grakn.Transaction.Type transactionType) {
+            this.database = database;
+            this.sessionType = sessionType;
+            this.transactionType = transactionType;
+        }
+        public String database() { return database; }
+        public Grakn.Session.Type sessionType() { return sessionType; }
+        public Grakn.Transaction.Type transactionType() { return transactionType; }
+    }
+    public DatabaseCreate asDatabaseCreate() { return (DatabaseCreate)this; }
+    public DatabaseDelete asDatabaseDelete() { return (DatabaseDelete)this; }
+    public Transaction asTransaction() { return (Transaction) this; }
+
+    public static ReplCommand getCommand(LineReader reader, Printer printer, String prompt) {
+        ReplCommand command = null;
+        while (command == null) {
+            String[] tokens = getTokens(reader, printer, prompt);
+            if (tokens.length == 1 && tokens[0].equals("exit")) {
```

Comment:
then you can use the `ReplCommand.Exit.name` as the `.equals()` here, which feels less like manually duplicated code.

`name` probably is the right variable anymore then - probably `syntax` or something

Reply:
The problem is there's no one-to-one mapping between keywords and the command. For example, `database list` command takes two keywords, whereas a query takes no keyword.